assembl/static/js/app/views/message.js

```diff
@@ -1524,6 +1545,52 @@ var MessageView = Marionette.LayoutView.extend({
     this.listenToOnce(this.messageModerationOptionsView, 'moderationOptionsClose', this.onModerationOptionsClose);
   },
 
+  onDeleteMessageClick: function(ev){
+    var that = this;
+    // We could try to minimize context switching for the user, by scrolling the viewport to the message the user wants to delete, as soon as the confirmation popin opens, using this line of code:
+    // that.messageListView.scrollToMessage(that.model, false, false);
+    
+    var onSubmit = function(ev){
+      var analytics = Analytics.getInstance();
+      analytics.trackEvent(analytics.events.MESSAGE_LIKED);
+      // we could use that.model.destroy() instead, and add || method == "delete" to models/message.js::sync()
+      var message_delete_url = that.model.getApiV2Url();
+      Promise.resolve(
+        $.ajax(message_delete_url, {
+          method: "DELETE",
+          contentType: "application/json",
+          dataType: "json"
+        })
+      ).then(function(data) {
+        Growl.showBottomGrowl(
+          Growl.GrowlReason.SUCCESS,
+          i18n.gettext('Message has been successfully deleted.'),
+          { delay: 12000 }
+        );
+
+        // Refresh the messageList
+        that.messageListView.render();
+        setTimeout(function(){
```

Comment:
@paolina92 I think this has to do with the scrollTo? Or maybe waiting for other components to render?